### PR TITLE
GEOPY-473

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-473 - Deal with groupOptional validation**

### CODE REVIEW

File: geoh5py/ui_json/utils.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+            val = fun(val)
+        else:
+            val = fun(val, *args)
+    return val
+
+
+def flatten(var: dict[str, Any]) -> dict[str, Any]:
+    """Flattens ui.json format to simple key/value pair."""
+    data: dict = {}
+    for key, value in var.items():
```
Comment (benk-mira) [inline geoh5py/ui_json/utils.py:63]:
```suggestion
    for name, value in var.items():
```

**Review by benk-mira — COMMENTED** [review]
Trading issues with Dom, stopping short here to push changes proposed in this review

File: tests/ui_json_test.py [hunk 2]
```diff
@@ -583,6 +596,20 @@ def test_input_file(tmp_path):
             raise ValueError(f"Input '{key}' differs from the output.")
 
 
+def test_write_ui_json(tmp_path):
+    # Make sure that none2str is applied in dict_mapper
+    workspace = get_workspace(tmp_path)
+    ui_json = deepcopy(default_ui_json)
+    ui_json["geoh5"] = workspace
+    ui_json["test"] = templates.float_parameter(optional="disabled")
```
Comment (domfournier) [inline tests/ui_json_test.py:604]: Can we change this whole business of "enabled" "disabled" to True or False instead? Just need to update the `optional_parameter` method in `templates`

**Review by domfournier — COMMENTED** [review]
Love it.

**Pull request merged**